### PR TITLE
Fix python numpy and pandas

### PR DESCRIPTION
**Pull request by coffee-cup**
## What does this PR address?
- Add the `zlib` and `stdenv.cc.cc.lib` Nix packages as libraries if numpy is detected
- Docker test for build and running python projects that use numpy and pandas

### CODE REVIEW

**Review by JakeCooper — APPROVED** [review]

File: src/providers/python.rs [hunk 1]
```diff
@@ -47,7 +47,15 @@ impl Provider for PythonProvider {
             pkgs.append(&mut vec![Pkg::new("postgresql"), Pkg::new("gcc")]);
         }
 
-        Ok(Some(SetupPhase::new(pkgs)))
+        let mut setup_phase = SetupPhase::new(pkgs);
+
+        // Numpy needs some C headers to be available
+        // stdenv.cc.cc.lib -> https://discourse.nixos.org/t/nixos-with-poetry-installed-pandas-libstdc-so-6-cannot-open-shared-object-file/8442/3
+        if PythonProvider::uses_numpy(app)? {
+            setup_phase.add_libraries(vec!["zlib".to_string(), "stdenv.cc.cc.lib".to_string()]);
```
Comment (Milo123459) [inline src/providers/python.rs:55]: Shouldn't this also add this to the nixpkgs to be installed?

File: src/providers/python.rs [hunk 2]
```diff
@@ -305,6 +313,14 @@ impl PythonProvider {
                 .ok_or_else(|| anyhow::anyhow!("failed to load pyproject.toml"))?),
         ))
     }
+
+    fn uses_numpy(app: &App) -> Result<bool> {
+        Ok(app.includes_file("requirements.txt")
+            && app
+                .read_file("requirements.txt")?
+                .to_lowercase()
+                .contains("numpy"))
```
Comment (Milo123459) [inline src/providers/python.rs:322]: This should probably support things like poetry.lock

**Review by Milo123459 — CHANGES_REQUESTED** [review]

File: src/providers/python.rs (hunk 2 above)
Comment (coffee-cup) [reply src/providers/python.rs:322]: ya good point?

File: src/providers/python.rs (hunk 1 above)
Comment (coffee-cup) [reply src/providers/python.rs:55]: No it is not necessary

Comment (Milo123459) [reply src/providers/python.rs:55]: Then what does add_libraries do? What is the difference?

File: src/providers/python.rs [hunk 3]
```diff
@@ -305,6 +313,22 @@ impl PythonProvider {
...
+    fn uses_numpy(app: &App) -> Result<bool> {
+        let requirements_numpy = app.includes_file("requirements.txt")
+            && app
+                .read_file("requirements.txt")?
+                .to_lowercase()
+                .contains("numpy");
+
+        let project_numpy = app.includes_file("pyproject.toml")
+            && app
+                .read_file("pyproject.toml")?
```
Comment (Milo123459) [inline src/providers/python.rs:326]: @coffee-cup This should have also checked poetry.lock

**Pull request merged**